### PR TITLE
[DOC] Updates to the overview doc + User Quotas

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**

Updated the Overview to cover:

- KafkaConnector
- PKCS #12
- Kafka listener certificates

Plus, for Using Guide:

- New KafkaUser section to describe user quotas
- Edit of KafkaUserQuotas description

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [x ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by PaulRMellor** [comment]
I found the current layout of chapter 6 Using the User Operator a bit confusing,
so I've made a note to rework it

File: documentation/modules/overview/con-configuration-points-broker.adoc [hunk 1]
```diff
@@ -43,6 +43,8 @@ External listeners expose Kafka by specifying a `type`:
 * `ingress` to use Kubernetes _Ingress_ and the {NginxIngressController}.
 --
 
+If you are using xref:security-configuration-authentication_{context}[{oauth} for token-based authentication], you can configure listeners to connect to the authorization server.
```
Comment (scholzj) [inline documentation/modules/overview/con-configuration-points-broker.adoc]: This is a bit weird ... _configure listeners to connect to_. I think we should try to find something better. MAybe _to **use** the authentication server_?

File: documentation/modules/overview/con-configuration-points-common.adoc [hunk 2]
```diff
@@ -14,6 +14,8 @@ Bootstrap servers:: Bootstrap servers are used for host/port connection to a Kaf
 * Kafka Bridge
 * Kafka MirrorMaker producers and consumers
 CPU and memory resources:: You request CPU and memory resources for components. Limits specify the maximum resources that can be consumed by a given container.
++
+When you define the specification for your Kafka cluster using the `Kafka` resource, you can set resource usage limits for the Topic Operator and User Operator.
```
Comment (scholzj) [inline documentation/modules/overview/con-configuration-points-common.adoc]: Thsi sounds a bit weird. Can't we just say _Resource requests and limits for TO and UO can be set in the Kafka CR_?

File: documentation/modules/overview/con-configuration-points-connect.adoc [hunk 3]
```diff
@@ -56,3 +56,45 @@ To create the image, you can use:
```
Comment (laidan6000) [inline documentation/modules/overview/con-configuration-points-connect.adoc]: ^
certain type of external system > certain types of external systems

File: documentation/modules/overview/con-configuration-points-connect.adoc [hunk 4]
```diff
@@ -56,3 +56,45 @@ To create the image, you can use:
...
+spec:
+  class: org.apache.kafka.connect.file.FileStreamSourceConnector
+  tasksMax: 2
+  config:
+    file: "/opt/kafka/LICENSE"
+    topic: my-topic
+    # ...
+----
+
+You enable `KafkaConnector` by adding an annotation to the `KafkaConnect` resource.
```
Comment (laidan6000) [inline documentation/modules/overview/con-configuration-points-connect.adoc]: KafkaConnector > KafkaConnectors

File: documentation/modules/overview/con-configuration-points-connect.adoc [hunk 5]
```diff
@@ -56,3 +56,45 @@ To create the image, you can use:
 
 * A Kafka container image on {DockerRepository} as a base image
 * OpenShift {docs-okd} and the {docs-okd-s2i} framework to create new container images
+
+[discrete]
+== Managing connectors
+
+You can use the `KafkaConnector` resource or the link:https://kafka.apache.org/documentation/#connect_rest[Kafka Connect API] to create and manage connector instances in a Kafka Connect cluster.
```
Comment (laidan6000) [inline documentation/modules/overview/con-configuration-points-connect.adoc]: Please change to "Kafka Connect REST API" for consistency with the Using guide.

File: documentation/modules/overview/con-configuration-points-connect.adoc [hunk 6]
```diff
@@ -56,3 +56,45 @@ To create the image, you can use:
...
+The `KafkaConnector` resource offers a Kubernetes-native approach, and is managed by the Cluster Operator.
+
+The `spec` for the `KafkaConnector` resource specifies the connector class and configuration settings, as well as the maximum number of connector _tasks_ to handle the data.
+
+[discrete]
+== Example YAML showing KafkaConnector configuration
+[source,yaml,subs="attributes+"]
+----
+apiVersion: kafka.strimzi.io/v1alpha1
```
Comment (laidan6000) [inline documentation/modules/overview/con-configuration-points-connect.adoc]: `{KafkaApiVersion}` attribute as below?

(Not sure I've been using this attribute consistently either.)

File: documentation/modules/ref-kafka-user.adoc [hunk 7]
```diff
@@ -213,8 +220,28 @@ If a user is added to a list of super users in a Kafka broker configuration, the
 
 For more information on configuring super users, see xref:assembly-kafka-authentication-and-authorization-deployment-configuration-kafka[authentication and authorization] of Kafka brokers.
 
-== Additional resources
+== User quotas
+
+You can configure `spec` for the `KafkaUser` resource to enforce quotas so that a user does not exceed access to Kafka brokers based on a byte threshold or a time limit of CPU utilization.
```
Comment (laidan6000) [inline documentation/modules/ref-kafka-user.adoc]: spec > the spec

**Pull request merged**